zstacklib/zstacklib/test/test_safe_delete.py

```
@@ -0,0 +1,299 @@
+_start = _src.index('\nblack_dpath_list = ')
+# Find end: after "def safe_delete_paths" ... "return failed"
+# We need everything from black_dpath_list through safe_delete_paths
+_end_marker = '\ndef rm_file_checked('
+_end = _src.index(_end_marker, _start)
+
+_fragment = _src[_start:_end]
+
+# Provide stubs for dependencies used in the fragment
+import logging as _logging
+_ns['logger'] = _logging.getLogger('test_safe_delete')
+_ns['_string_types'] = (str,) if sys.version_info[0] >= 3 else (str, unicode)
```

Ruff (0.15.9)

[error] Undefined name `unicode`

(F821)